examples/bfetch_explorer/tsconfig.json

```diff
@@ -13,6 +13,7 @@
   ],
   "exclude": [],
   "references": [
-    { "path": "../../src/core/tsconfig.json" }
+    { "path": "../../src/core/tsconfig.json" },
+    { "path": "../../src/plugins/kibana_react/tsconfig.json" },
```

Comment:
it's not necessary, as we don't build the project with `tsc`, but it reduces the number of processed files
*with refs*
with refs
```
Files:                         884
Lines:                      219774
Nodes:                      789226
Identifiers:                286286
```
without refs
```
Files:                        1214
Lines:                      271616
Nodes:                      928343
Identifiers:                333268
```

Reply:
What project did you build for these stats?

There are a lot of other example projects that import kibana_react types, should we add references from their tsconfig.json files too?
<img width="693" alt="Screenshot 2020-09-25 at 00 35 32" src="https://user-images.githubusercontent.com/1281646/94207128-3affa180-fec7-11ea-9e1c-2d6246a7c947.png">

Reply:
It might become hard to maintain all the references if we start having many small projects. Maybe we can set `composite: true` in `./tsconfig.refs.json` and then all other projects that might need these references just contain a single reference to `./tsconfig.refs.json` instead of a reference for each project.

I didn't test this, but my understanding is that you can create such hierarchies of projects and TS will still be clever about what files it includes in the build instead of building the whole parent project.

Reply:
> I didn't test this, but my understanding is that you can create such hierarchies of projects and TS will still be clever about what files it includes in the build instead of building the whole parent project.

Tested just to make sure it doesn't work: TS cannot build a DAG since a project references itself and other project dependent on it. 

> It might become hard to maintain all the references if we start having many small projects.

I declared the reference to `kibana_react` only for projects with an **explicit** dependency on `kibanaReact` plugin in `kibana.json`. Indeed, `kibana_react` might me compiled as a transitive dependency (you can run `tsc` with `--traceResolution` flag to learn how a file was included). But we will resolve the over-compilation problem when switching all the plugins to TypeScript project refs, so I think it's okay for now.